src/streaming-services/nrk/NrkEvents.ts

```diff
@@ -0,0 +1,115 @@
+import { registerScrobbleEvents } from '../common/common';
+import { ScrobbleEvents } from '../common/ScrobbleEvents';
+import { NrkApi } from './NrkApi';
+
+class _ScrobblerTemplateEvents extends ScrobbleEvents {
+	progress = 0;
+	videoId = '';
+	playbackStarted = false;
+	url = '';
+	setBeforeUnloadEvent = false;
+
+	checkForChanges = async (): Promise<void> => {
+		const player = await NrkApi.getSession();
+		if (typeof player !== 'undefined') {
+			if (player) {
+				if (this.videoId !== player.mediaItem.id) {
+					console.warn('new sessinoID', player.mediaItem.id);
+					if (this.isPlaying || this.playbackStarted) {
+						console.warn('stopping current scrobbling');
+						await this.stop();
+						this.isPlaying = false;
+					}
+					if (player.playbackStarted && !player.sequenceObserver.isPaused) {
+						console.warn('start new scrobbling');
+						await this.start();
+						this.isPlaying = true;
+					}
+					this.videoId = player.mediaItem.id;
+					this.isPaused = false;
+					this.playbackStarted = player.playbackStarted;
+				} else if (player.playbackStarted) {
+					if (player.sequenceObserver.isPaused) {
+						if (!this.isPaused) {
+							console.warn('pause');
+							await this.pause();
+						}
+					} else {
+						if (!this.isPlaying) {
+							console.warn('start');
+							await this.start();
+						}
+					}
+					this.isPaused = player.sequenceObserver.isPaused;
+					this.isPlaying = !player.sequenceObserver.isPaused;
+					this.playbackStarted = player.playbackStarted;
+				}
+
+				if (this.isPlaying) {
+					const newProgress = Math.round((player.currentTime / player.duration) * 10000) / 100;
+					if (this.progress !== newProgress) {
+						await this.updateProgress(newProgress);
+						this.progress = newProgress;
+					}
+				}
+				if (!this.setBeforeUnloadEvent) {
+					// eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
This is technically not necessary when closing a tab, because we already have this: https://github.com/trakt-tools/universal-trakt-scrobbler/blob/master/src/modules/background/background.ts#L118 which should take care of stopping scrobbles when tabs close. Did you test it and it didn't work?

As for navigating to the front page, can't you just check that in the `checkForChanges` function?

Reply:
This actually fixed some other issues now.